src/websocket_service/handlers/notify_subscribe.rs

```diff
@@ -48,27 +58,63 @@ pub async fn handle(
         hex::encode(client_pubkey),
         project_data.signing_keypair.private_key,
     )?;
-    info!("[{request_id}] Response_sym_key: {}", &response_sym_key);
+    info!("[{request_id}] Response_sym_key: {}", &response_sym_key); // TODO don't log this
 
     let msg: NotifyMessage<NotifySubscribe> = decrypt_message(envelope, &response_sym_key)?;
 
     info!("[{request_id}] Register message: {:?}", &msg);
 
     let id = msg.id;
 
-    let sub_auth = SubscriptionAuth::from_jwt(&msg.params.subscription_auth)?;
+    let sub_auth = from_jwt::<SubscriptionAuth>(&msg.params.subscription_auth)?;
     let sub_auth_hash = sha256::digest(msg.params.subscription_auth);
 
-    // TODO verify `sub_auth.aud` matches `project_data.identity_keypair` (?)
+    if sub_auth.act != "notify_subscription" {
+        return Err(AuthError::InvalidAct)?;
+    }
+
+    // TODO verify `sub_auth.iss` matches SOMETHING (blockchain account?) because
+    // otherwise what's the purpose of having a signature
```

Comment:
iss is Identity Key. here you would call `verify_identity()`

Reply:
Ahhhhh ok, that makes sense. I'll do that in a future PR as we discussed :)